docs/sources/reference/plugin-mvp.md

```diff
@@ -0,0 +1,87 @@
+Brian Goff and Luke Marsden paired on an MVP of the plugin handshake interface.
+
+# pre-MVP design
+
+We ended up with the following design:
+
+```
+$ docker run --plugin \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    clusterhq/flocker-plugin
+```
+
+Note the new `--plugin` argument.
+
+Internally this translates into:
+
+```
+$ docker run -d \
+	-v /var/lib/docker/containers/<container_id>/plugin/:/var/run/docker/ \
+	-v /var/run/docker.sock:/var/run/docker.sock \
+	clusterhq/flocker-plugin
+```
+
+Docker blocks on waiting for a successful handshake with the plugin on `/var/run/docker/plugin.sock` with an HTTP `POST /v1/handshake` with an empty body.
+
+# Plugin API
+
+## `POST /v1/handshake`
+
+**Request:** POST with empty body
+
+**Response:** `application/json` as follows:
+
+```
+{
+ InterestedIn: ["volume"],
+ Name: "flocker",
+ Author: "Luke Marsden <luke@clusterhq.com>",
+ Org: "ClusterHQ, Inc.",
+ Website: "https://clusterhq.com/",
+}
+```
+
+`PluginName` is a human readable short string identifying the plugin.
+
+`InterestedIn` is a list of extension points.
+
+Extension points currently supported are:
+
+* `volume` - called when a volume is bind-mounted into a container at create or start time
```

Comment:
I assume we can replace "when" with "after" ?  just to be clear about when it is called.